quarkchain/cluster/miner.py

```diff
@@ -254,8 +257,12 @@ def _mine_new_block_async(self):
 
         if now is None:  # clock open for mock
             now = time.time()
-        # 5 sec interval magic number
-        if not self.current_work or now - self.current_work.header.create_time > 5:
+        tip = self.get_header_tip_func()
+        if (
+            not self.current_work  # no cache
+            or self.current_work.header.height <= tip.height  # cache outdated
```

Comment:
if tip is reset by root block, shard chain's tip may have smaller height.  I think we should check

or self.current_work.header.prev_has != tip.get_hash()

to make sure we are working on the correct tip + 1 block.

Reply:
done